src/assisted_test_infra/test_infra/helper_classes/cluster.py

```diff
@@ -1292,3 +1293,79 @@ def wait_and_kill_installer(self, host):
         # Kill installer to simulate host error
         selected_node = self.nodes.get_node_from_cluster_host(host)
         selected_node.kill_installer()
+
+    def format_host_current_network(self):
+        """
+        return:
+        host_network["mac_address"] = {
+                = [],
+            ipv4_addresseses = []
+            ipv4_addresseses = []
+        }
+        """
+        hosts_list = self.api_client.get_cluster_hosts(cluster_id=self._config.cluster_id)
+        log.debug(f"hosts list is: {hosts_list}")
+        host_network = {}
+        for host in hosts_list:
+            log.debug(f"Validation host {host['requested_hostname']}")
+            inventory = yaml.safe_load(host["inventory"])
+
+            interfaces = inventory["interfaces"]
+            for interface in interfaces:
+                for _, address_version in consts.IP_VERSIONS.items():
+                    if address_version in interface.keys():
+                        if len(interface[address_version]) != 0:
+                            host_network[interface["mac_address"]] = {address_version: interface[address_version]}
+        log.debug(f"host_network {host_network}")
+        return host_network
+
+    def validate_static_ip(self):
+        if self._infra_env_config.static_network_config is None:
+            log.debug("Skipping static IP validation")
+            return
+
+        log.info("Starting static IP validation")
+        self.wait_until_hosts_are_discovered()
+
+        host_network = self.format_host_current_network()
+        if host_network == {}:
+            raise Exception("Couldn't find host network configurations")
+
+        host_failure = []
+        for expected_host_mapping in yaml.safe_load(str(self._infra_env_config.static_network_config)):
+            for expected_host_interfaces in expected_host_mapping["mac_interface_map"]:
+                expected_interface_name, expected_interface_mac = (
+                    expected_host_interfaces["logical_nic_name"],
+                    expected_host_interfaces["mac_address"],
+                )
+            log.debug(f"Validating expected interface {expected_interface_name} with mac {expected_interface_mac}")
+            if expected_interface_mac not in host_network.keys():
+                host_failure.append(f"missing mac address {expected_interface_mac}")
+                continue
+
+            network_yaml = yaml.safe_load(return_yaml_from_string(expected_host_mapping["network_yaml"]))
+            interfaces = network_yaml["interfaces"]
+            expected_hosts_address = []
+            for current_interface in interfaces:
```

Comment:
One suggestion to improve the algorithm, build a similar map returned by `format_host_current_network` but from `network_yaml`. And then diff the maps from the inventory and the map from network_yaml (python provides the libdiff that I think could be leveraged).